src/main/java/com/blackducksoftware/integration/hub/dataservice/user/UserDataService.java

```diff
@@ -71,4 +72,16 @@ public UserDataService(final RestConnection restConnection, final UserRequestSer
         return resolvedProjectViews;
     }
 
+    public List<RoleView> getRolesForUser(final String userName) throws IntegrationException {
+        final UserView user = userRequestService.getUserByUserName(userName);
+        return getRolesForUser(user);
+    }
+
+    public List<RoleView> getRolesForUser(final UserView userView) throws IntegrationException {
+        final String userRolesLink = metaService.getFirstLink(userView, MetaService.ROLES_LINK);
```

Comment:
Can you move this code into the UserRequestService? We typically have the requests handled by the request services.